go/tools/gazelle/rules/generator.go

```diff
@@ -91,15 +91,25 @@ func (g *generator) Generate(rel string, pkg *build.Package) ([]*bzl.Rule, error
 		rules = append(rules, p)
 	}
 
-	libName := defaultLibName
-	r, err := g.generateLib(rel, libName, pkg)
+	cgoLibrary := ""
+	if len(pkg.CgoFiles) != 0 || len(pkg.CFiles) != 0 || len(pkg.HFiles) != 0 {
```

Comment:
does it ever make sense to have a cgo_library with no CgoFiles but CFiles?
Also, what about CXXFiles? (which you already appear to handle in generateCgoCLib)

e.g. if len(CgoFiles) == 0 but any of C/CXX/H are >0, should gazelle just print a warning and skip cgo?